src/flows/Onboarding/OnboardingCreateReserve.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import { useCreateReserve } from './hooks';
+import { Button } from '@/src/components/ui/button';
+import { mutationToPromise } from '@/src/lib/mutations';
+import { SuccessResponse } from '@/src/client';
+import { useOnboardingContext } from './context';
+
+type OnboardingCreateReserveProps = PropsWithChildren<
+  ButtonHTMLAttributes<HTMLButtonElement> & {
+    onSuccess?: (data: SuccessResponse) => void | Promise<void>;
+    onError?: (error: unknown) => void;
+    onSubmit?: () => void | Promise<void>;
+  }
+>;
+
+export function OnboardingCreateReserve({
+  onSubmit,
+  onSuccess,
+  onError,
+  ...props
+}: OnboardingCreateReserveProps) {
+  const { onboardingBag } = useOnboardingContext();
+  const createReserveMutation = useCreateReserve();
+
+  const { mutateAsync: createReserveMutationAsync } = mutationToPromise(
+    createReserveMutation,
+  );
+
+  const handleSubmit = async () => {
+    try {
+      await onSubmit?.();
+
+      if (!onboardingBag.employmentId) {
+        throw new Error('Employment ID is required');
+      }
+
+      const response = await createReserveMutationAsync({
```

Comment:
As a safety measure we should only allow running `createReserveMutationAsync` only if `creditRiskStatus === 'deposit_required'`

And the `OnboardingInvite` component should also be update with a similar verification, probably `creditRiskStatus !== 'deposit_required'`